compiler/src/org.graalvm.compiler.phases.common/src/org/graalvm/compiler/phases/common/AddressLoweringByUsePhase.java

```diff
@@ -94,8 +94,13 @@ protected void run(StructuredGraph graph) {
             // the lowered address amy already be a replacement
             // in which case we want to use it not delete it!
             if (lowered != address) {
-                address.replaceAtUsages(lowered);
-                GraphUtil.killWithUnusedFloatingInputs(address);
+                // replace original with lowered at this usage only
+                // n.b. lowered is added unique so repeat lowerings will elide
```

Comment:
How about "n.b. since lowered was added to the graph with Graph.unique, all lowerings of an address node for the same datum type will common to the same node".